<a href="https://colab.research.google.com/github/BerniceJewel/Machine-Learning-with-Applications-in-Finance---Linear-Regression/blob/main/Machine_Learning_with_Applications_in_Finance_Linear_Regression.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Chap. 3: **Linear Regression** Q.10 (Section a-g)

**10. This question should be answered using the **Carseats** data set.**


In [50]:
pip install ISLP

In [51]:
import os
import numpy as np
import pandas as pd
import seaborn as sns
import statsmodels.api as sm
import matplotlib.pyplot as plt
from ISLP.models import (ModelSpec as MS, summarize)

In [52]:
from ISLP import load_data
Carseats = load_data('Carseats')
Carseats.columns

Index(['Sales', 'CompPrice', 'Income', 'Advertising', 'Population', 'Price',
       'ShelveLoc', 'Age', 'Education', 'Urban', 'US'],
      dtype='object')

In [53]:
print(Carseats.head())


   Sales  CompPrice  Income  Advertising  Population  Price ShelveLoc  Age  \
0   9.50        138      73           11         276    120       Bad   42   
1  11.22        111      48           16         260     83      Good   65   
2  10.06        113      35           10         269     80    Medium   59   
3   7.40        117     100            4         466     97    Medium   55   
4   4.15        141      64            3         340    128       Bad   38   

   Education Urban   US  
0         17   Yes  Yes  
1         10   Yes  Yes  
2         12   Yes  Yes  
3         14   Yes  Yes  
4         13   Yes   No  


In [54]:
Carseats = Carseats.astype({'ShelveLoc': 'category', 'Urban': 'category', 'US': 'category'})

In [55]:
Carseats.dtypes

Sales           float64
CompPrice         int64
Income            int64
Advertising       int64
Population        int64
Price             int64
ShelveLoc      category
Age               int64
Education         int64
Urban          category
US             category
dtype: object

 **a.   Fit a multiple regression model to predict **Sales** using ***Price***,
 ***Urban***, and ***US***.**


In [56]:
#Fit the model

design = MS(['Price', 'Urban', 'US'])
X = design.fit_transform(Carseats)
y = Carseats['Sales']

In [57]:
model = sm.OLS(y, X)
results = model.fit()
summarize(results)

coef  std err       t  P>|t|
intercept   13.0435    0.651  20.036  0.000
Price       -0.0545    0.005 -10.389  0.000
Urban[Yes]  -0.0219    0.272  -0.081  0.936
US[Yes]      1.2006    0.259   4.635  0.000

**b.   Provide an interpretation of each coefficient in the model. Be careful—some of the variables in the model are qualitative!**





1. The intercept **13.0435** represents the number of car seats sold on average when all other predictors are disregarded.
   
2. The `Price` coefficient is negative and so sales will fall by roughly 545 seats(0.0545x1000)for every unit($1) increase in price.

3. The `Urban=Yes` coeff is not **statistically significant**.Stores in urban locations sell 0.0219 fewer units (in thousands) than average. However we note the p-value is very high **(0.936)**indicating there is no statistical evidence of a difference in average units sold between urban and rural stores.
   
4. The `US=Yes` coeff is 1.2, and this means an average increase in car seat sales of (1.2006x1000)=12006 units when `US=Yes`


 **c. Write out the model in equation form, being careful to handle
 the qualitative variables properly.**

**Linear regression equation**




$$y = 13.0435\ + -0.0545X_2 \ + -0.0219X_2 \ + 1.20006X_3$$


$$X_1\ =Price ,\\ X_2\ =Urban[YES] \ and \ 0\ if\ Rural, \ 1\  if\ Urban,  \\  X_3\ = US[YES]  \ 0\ if\  Store \ is\  Not\  in \ US,  \ 1\  if\ it\  is\  in\  US.$$






**d.   For which of the predictors can you reject the null hypothesis
 $$H\ _0 : β\ _ j =0$$**


We can reject the null hypothesis for the predictors Price and US[Yes].
This implies that their effect on sales is statistically significant.



**e. fit a smaller model that only uses the predictors for which there is evidence of association with the outcome.**



In [58]:
design = MS(['Price', 'US'])
X = design.fit_transform(Carseats)
y = Carseats['Sales']

In [59]:
model = sm.OLS(y, X)
results1 = model.fit()
summarize(results1)

coef  std err       t  P>|t|
intercept  13.0308    0.631  20.652    0.0
Price      -0.0545    0.005 -10.416    0.0
US[Yes]     1.1996    0.258   4.641    0.0



**f. Compare the fit of the two models**




In [60]:
summarize (results)

coef  std err       t  P>|t|
intercept   13.0435    0.651  20.036  0.000
Price       -0.0545    0.005 -10.389  0.000
Urban[Yes]  -0.0219    0.272  -0.081  0.936
US[Yes]      1.2006    0.259   4.635  0.000

In [61]:
summarize(results1)

coef  std err       t  P>|t|
intercept  13.0308    0.631  20.652    0.0
Price      -0.0545    0.005 -10.416    0.0
US[Yes]     1.1996    0.258   4.641    0.0

In [62]:
sm.stats.anova_lm(results, results1)
#Dropping the predictor Urban[Yes] does not seem to change the fit of the model to the data.

df_resid          ssr  df_diff  ss_diff         F  Pr(>F)
0     396.0  2420.834671      0.0      NaN       NaN     NaN
1     397.0  2420.874462     -1.0 -0.03979  0.006525     NaN



**g.   95% confidence interval for the coefficients**



In [63]:
results1 = results1.conf_int()
results1

0          1
intercept  11.79032  14.271265
Price      -0.06476  -0.044195
US[Yes]     0.69152   1.707766